lib/vmpooler/api/v1.rb

```diff
@@ -400,8 +423,16 @@ def extract_templates_from_query_params(params)
       content_type :json
       result = { 'ok' => false }
 
-      if payload and !payload.empty?
-        result = atomically_allocate_vms(payload)
+      if payload
+        empty = payload.delete('empty')
+        invalid = payload.delete('invalid')
+        statsd.increment(statsd_prefix + '.checkout.empty', empty) if !empty.nil?
+        statsd.increment(statsd_prefix + '.checkout.invalid', invalid) if !invalid.nil?
+        if !payload.empty?
+          result = atomically_allocate_vms(payload)
+        else
+          status 404
+        end
```

Comment:
See like, the last 3 or 4 comments re: simplifying some of lines 426 through 435 (`unless`, `#nil?`, truthiness, etc.).